### PR TITLE
Fix unnecessary includes in code generated headers

### LINKED ISSUE #3991: C++ don't leak headers from _ext impls into public headers
Most notoriously, Asset3D leaks <fstream> and <filesystem> into rerun.hpp which is quite heavy!

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Fixes #3991
* Allow several _ext -> header copy blocks
* no longer automatically add all includes from _ext file to header
* detect includes in copy to header blocks and add them to the header list
* rename `util.hpp` to `warning_macros.hpp`
* change copy-to-codegen markers to an xml like syntax, shortening it a bit and making it easier to read

Oh and every _ext file I touched a bit more I cleaned up to use a less autocomplete friendly but easier to understand way of handling the extension code. Let's prefer this going forward (there's no functional difference though).

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/4132) (if applicable)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4132)
- [Docs preview](https://rerun.io/preview/2799f8838b5920d4343011d09b6a75ce47666355/docs) 
- [Examples preview](https://rerun.io/preview/2799f8838b5920d4343011d09b6a75ce47666355/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: crates/re_types_builder/src/codegen/cpp/mod.rs [hunk 1]
```diff
@@ -207,50 +207,56 @@ fn hpp_type_extensions(
...
-                }
-            } else if let Some(start) = line.find('<') {
-                let end = line.rfind('>').unwrap_or_else(|| {
-                    panic!(
+    let mut remaining_content = &content[..];
+    let mut hpp_extension_string = String::new();
+
+    while let Some(start) = remaining_content.find(COPY_TO_HEADER_START_MARKER) {
+        let end = remaining_content.find(COPY_TO_HEADER_END_MARKER).unwrap_or_else(||
+            panic!("C++ extension file has a start marker but no end marker. Expected to find '{COPY_TO_HEADER_START_MARKER}' in {extension_file:?}")
```
Comment (jleibs) [inline crates/re_types_builder/src/codegen/cpp/mod.rs:218]:
```suggestion
            panic!("C++ extension file has a start marker but no end marker. Expected to find '{COPY_TO_HEADER_END_MARKER)}' in {extension_file:?}")
```

File: crates/re_types_builder/src/codegen/cpp/mod.rs [hunk 2]
```diff
@@ -207,50 +207,56 @@ fn hpp_type_extensions(
...
+
+        let extensions = &remaining_content[start + COPY_TO_HEADER_START_MARKER.len()..end];
+
+        // Comb through any includes in the extension string.
+        for line in extensions.lines() {
+            if line.starts_with("#include") {
+                if let Some(start) = line.find('\"') {
+                    let end = line.rfind('\"').unwrap_or_else(|| {
+                        panic!(
+                            "Expected to find '\"' include line {line} in file {extension_file:?}"
```
Comment (jleibs) [inline crates/re_types_builder/src/codegen/cpp/mod.rs:232]:
```suggestion
                            "Expected to find ending '\"' in include line {line} in file {extension_file:?}"
```

File: rerun_cpp/src/rerun/archetypes/asset3d_ext.cpp [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
+#include <filesystem>
+
+// </CODEGEN_COPY_TO_HEADER>
 
 namespace rerun {
     namespace archetypes {
 
-#ifdef EDIT_EXTENSION
-        // [CODEGEN COPY TO HEADER START]
+#if 0
```
Comment (jleibs) [inline rerun_cpp/src/rerun/archetypes/asset3d_ext.cpp:18]: Why not use `EDIT_EXTENSION` here?

Comment (Wumpf) [reply rerun_cpp/src/rerun/archetypes/asset3d_ext.cpp:18]: I'm phasing out the EDIT_EXTENSION way of doing things. The idea was to have LSP friendly code but that requires a bit of extra maintenance and everyone editing these (myself included) gets this wrong all the time. Also, it gave the impression that there's some additional magic at work.
So instead I make these being honest now and just explicitly disable stuff.

**Pull request merged**